README.md

```diff
@@ -685,7 +685,8 @@ If you're using the `X-SourceMap` header instead, you can just omit `sourceMap.u
   pass `pure_funcs: [ 'Math.floor' ]` to let it know that this
   function won't produce any side effect, in which case the whole
   statement would get discarded.  The current implementation adds some
-  overhead (compression will be slower).
+  overhead (compression will be slower). Make sure symbols under `pure_funcs`
+  are also under `mangle.reserved` to avoid mangling.
```

Comment:
Why not do that automatically?

Reply:
Too much hassle to maintain for a non-default feature.

Reply:
This is actually a regression from `2.x`. The `pure_funcs` used to work correctly without `mangle.reserved`:

https://github.com/mishoo/UglifyJS2/issues/3065#issuecomment-379941419

Further, unlike `mangle.reserved`, you will notice in the link that the retained `pure_funcs` were also mangled after they were processed in `2.x` as one would expect.

If you want to correct the comment above just erase the note about using `mangle.reserved` and say to use `--no-rename` as a workaround in `3.x` when mangling with `pure_funcs`. Alternatively, the specified `pure_funcs` could be excluded in the `rename` pass.  Or simpler still the `rename` pass could just be disabled automatically when `pure_funcs` is specified.